posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -93,255 +100,89 @@ def parse_clickhouse_timestamp(s: str, tzinfo: timezone = timezone.utc) -> datet
-        _latest_merged_at: A timestamp representing an upper bound for events to squash. Obtained
-            as the latest timestamp of a person merge.
+        delete_grace_period_seconds: Number of seconds until an override can be deleted. This grace
+            period works on top of checking if the override was applied to all partitions. Defaults
+            to 24h.
+        dry_run: Do not run the queries when True.
     """
 
     partition_ids: list[str] = field(default_factory=list)
     team_ids: list[int] = field(default_factory=list)
-    person_overrides_to_delete: list[SerializablePersonOverrideToDelete] = field(default_factory=list)
     dictionary_name: str = "person_overrides_join_dict"
+    delete_grace_period_seconds: int = 24 * 3600
     dry_run: bool = True
-    wait_for_mutations: bool = False
-    _latest_created_at: str | datetime | None = None
-
-    def __post_init__(self) -> None:
-        if isinstance(self._latest_created_at, datetime):
-            self.latest_created_at = self._latest_created_at
-
-    @property
-    def latest_created_at(self) -> datetime | None:
-        if isinstance(self._latest_created_at, str):
-            return datetime.fromisoformat(self._latest_created_at)
-        return self._latest_created_at
-
-    @latest_created_at.setter
-    def latest_created_at(self, v: datetime | str | None) -> None:
-        if isinstance(v, datetime):
-            self._latest_created_at = v.isoformat()
-        else:
-            self._latest_created_at = v
-
-    def iter_person_overides_to_delete(self) -> Iterable[SerializablePersonOverrideToDelete]:
-        """Iterate over SerializablePersonOverrideToDelete ensuring they are of that type.
-
-        Looking at the types, this seems pointless, just iterate over person_overrides_to_delete!
-        However, as Temporal passes inputs to and from activities, namedtuples will be cast to
-        lists. This method thus exists to transform them back into namedtuples.
-        """
-        for person_override_to_delete in self.person_overrides_to_delete:
-            yield SerializablePersonOverrideToDelete(*person_override_to_delete)
 
 
 @activity.defn
-async def prepare_person_overrides(inputs: QueryInputs) -> None:
+async def optimize_person_distinct_id_overrides(inputs: QueryInputs) -> None:
     """Prepare the person_overrides table to be used in a squash.
 
     This activity executes an OPTIMIZE TABLE query to ensure we assign the latest overrides for each old_person_id.
     """
     from django.conf import settings
 
-    activity.logger.info("Preparing person_overrides table for squashing")
-
-    optimize_query = "OPTIMIZE TABLE {database}.person_overrides ON CLUSTER {cluster} FINAL SETTINGS mutations_sync = 2"
+    optimize_query = "OPTIMIZE TABLE {database}.person_distinct_id_overrides ON CLUSTER {cluster} FINAL"
 
     if inputs.dry_run is True:
         activity.logger.info("This is a DRY RUN so nothing will be detached or optimized.")
```

Comment:
```suggestion
        activity.logger.info("This is a DRY RUN so nothing will be optimized.")
```

… though, I guess technically nothing will be detached, too. 😁

Reply:
😅